src/coreclr/jit/simd.cpp

```diff
@@ -663,7 +743,77 @@ CorInfoType Compiler::getBaseJitTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeH
                         getClassNameFromMetadata(baseTypeHnd, nullptr));
 
 #if defined(TARGET_XARCH)
-                if (strcmp(className, "Vector256`1") == 0)
+                if (strcmp(className, "Vector512`1") == 0)
```

Comment:
Not something to handle in this PR, but we have a lot of code duplication here that could probably be simplified down to a templated method that takes a pointer to the first handle for a given `Vector###`

Reply:
Likewise, if we reordered the handles to match the `CORINFO_TYPE_` order, we could avoid the switch tables.